eng/Versions.props

```diff
@@ -52,7 +52,7 @@
     <MicrosoftExtensionsOptionsConfigurationExtensionVersion>8.0.0</MicrosoftExtensionsOptionsConfigurationExtensionVersion>
     <MicrosoftExtensionsOptionsVersion>8.0.0</MicrosoftExtensionsOptionsVersion>
     <MicrosoftExtensionsPrimitivesVersion>8.0.0</MicrosoftExtensionsPrimitivesVersion>
-    <MicrosoftNetCompilersToolsetVersion>4.11.0-2.24270.4</MicrosoftNetCompilersToolsetVersion>
+    <MicrosoftNetCompilersToolsetVersion>4.12.0-3.24473.3</MicrosoftNetCompilersToolsetVersion>
```

Comment:
Curious: what if we just delete this and move to the compiler that comes with the .NET SDK.

Reply:
I can try that. But generally we sometimes want to consume new compiler features before we update our SDK, right? So then we would need to re-add this again.

Reply:
Although I updated this to bring in the "better conversion from collection expression" feature that's needed because of some string.Join ambiguities - and that might not be even in RC1, right? Well, definitely it's not in preview 7 which we are currently at.